### DISCUSSION TITLE
How to use loading template on Reflex

### DISCUSSION THREAD

**Discussion by kumaraguru1735** [reflex-dev/reflex#2455] [General]
How to use loading template on Reflex

**Comment by picklelo** [comment]
What do you mean loading template? Like something to show before the frontend connects to the backend?

**Reply by kumaraguru1735** [reply]
yes

**Reply by picklelo** [reply]
You can do something like this

```python
rx.cond(
    rx.State.is_hydrated,
    rx.text("my content"),
    rx.spinner(),
)
```

The `is_hydrated` flag checks whether the state has been loaded by the frontend yet, so you can embed your page in an `rx.cond` and show some sort of loading spinner while connecting.

We will update the main website docs with this info.
